### ISSUE TITLE
Cannot reach ICP from Mac host

### ISSUE THREAD

**Issue by mikespicer** [IBM/deploy-ibm-cloud-private#132]
After resolving a few problems that were answered by other issues I have hit a problem that I cannot resolve with being unable to reach the master from my mac host.  Ping gets no response, and all browsers are unable to reach the ICP console on port 8443.  

I changed the base_segment to 192.168.56 and when I ssh to the VM It does have an lxdbr0 entry in ifconfig that has the ip 192.168.56.100.  I don't see an entry for lxdbr0 in /etc/network/interfaces but am unsure if thats relevant.  There are also 2 workers that have addresses of 192.168.56.101 and 192.168.56.102 respectively.  /etc/hosts includes 192.168.56.100 resolved to master.icp and I can ping that name from the VM (not the mac).

Virtualbox shows that the VM has a host-only network adapter 'vboxnet1' with ipv4 address of 192.168.56.1 and mask of 255.255.255.0.  192.168.56.1, 192.168.56.101 and 192.168.56.102 do not respond to ping either.  

I can ping the mac host and worker containers from the VM.

Google searches have brought up issues like not having the host only network adapter configured and others that don't seem relevant.

Any ideas?

**Comment by mikespicer** [comment]
Some more info below and some questions based on these.  It looks like lsdbr0 is associated with eth1 but it looks like eth1 is associated with enp0s3 in /etc/network/interfaces and not enp0s8.  I thought that I had read that enp0s3 is for NAT allowing connections from the guest VM to the host Mac and that enp0s8 is for host only networking allowing the host MAC to reach the guest VM.  Do I need to associated lxdbr0 with an interface that is associated to enp0s8 somehow?

Details of lxdbr0
> vagrant@master:~$ lxc network show lxdbr0
config:
  bridge.driver: native
  bridge.external_interfaces: eth1
  bridge.mode: standard
  dns.domain: icp
  dns.mode: managed
  ipv4.address: 192.168.56.100/24
  ipv4.dhcp: "true"
  ipv4.dhcp.ranges: 192.168.56.100-192.168.56.254
  ipv4.firewall: "false"
  ipv4.nat: "true"
  ipv4.routing: "true"
  ipv6.address: none
  raw.dnsmasq: |
    dhcp-option-force=26,9000
    server=127.0.0.1
description: ""
name: lxdbr0
type: bridge
used_by:
\- /1.0/containers/worker1
\- /1.0/containers/worker2
managed: true
status: Created
locations:
\- none

Output from ifconfig (removed large number of calico entries)
>docker0   Link encap:Ethernet  HWaddr 02:42:a4:ef:2e:ae  
          inet addr:172.17.0.1  Bcast:172.17.255.255  Mask:255.255.0.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
>
>eth0      Link encap:Ethernet  HWaddr 08:00:27:67:d9:b9  
          inet addr:10.0.2.15  Bcast:10.0.2.255  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:7558149 errors:0 dropped:0 overruns:0 frame:0
          TX packets:2773635 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:7388602956 (7.3 GB)  TX bytes:155492310 (155.4 MB)
>
>eth1      Link encap:Ethernet  HWaddr 08:00:27:8a:c8:82  
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:24 errors:0 dropped:0 overruns:0 frame:0
          TX packets:643 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:7191 (7.1 KB)  TX bytes:205254 (205.2 KB)
>
>lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:1868901 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1868901 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1 
          RX bytes:952167929 (952.1 MB)  TX bytes:952167929 (952.1 MB)
>
>lxdbr0    Link encap:Ethernet  HWaddr 08:00:27:8a:c8:82  
          inet addr:192.168.56.100  Bcast:0.0.0.0  Mask:255.255.255.0
          UP BROADCAST RUNNING PROMISC MULTICAST  MTU:1500  Metric:1
          RX packets:656189 errors:0 dropped:0 overruns:0 frame:0
          TX packets:723303 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:77302085 (77.3 MB)  TX bytes:1552102345 (1.5 GB)
>
>tunl0     Link encap:IPIP Tunnel  HWaddr   
          inet addr:10.1.219.64  Mask:255.255.255.255
          UP RUNNING NOARP  MTU:1430  Metric:1
          RX packets:11335 errors:0 dropped:0 overruns:0 frame:0
          TX packets:6841 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1 
          RX bytes:4901877 (4.9 MB)  TX bytes:597691 (597.6 KB)
>
>veth9JKJWM Link encap:Ethernet  HWaddr fe:b6:82:7f:12:c8  
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:254285 errors:0 dropped:0 overruns:0 frame:0
          TX packets:283245 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:39006254 (39.0 MB)  TX bytes:501753645 (501.7 MB)
>
>vethIXXGVT Link encap:Ethernet  HWaddr fe:18:78:c5:e5:cc  
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:240940 errors:0 dropped:0 overruns:0 frame:0
          TX packets:264692 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:38753725 (38.7 MB)  TX bytes:456517108 (456.5 MB)

/etc/network/interfaces file
>vagrant@master:~$ more /etc/network/interfaces
\# This file describes the network interfaces available on your system
\# and how to activate them. For more information, see interfaces(5).
>
>source /etc/network/interfaces.d/*
>
>\# The loopback network interface
auto lo
iface lo inet loopback
>
>\# The primary network interface
auto enp0s3
iface enp0s3 inet dhcp
pre-up sleep 2
    post-up ifconfig eth1 promisc on
>
>iface eth1 inet manual

/etc/hosts file
>vagrant@master:~$ more /etc/hosts
127.0.0.1	localhost
192.168.56.100	master.icp	master
>
>\# The following lines are desirable for IPv6 capable hosts
::1     localhost ip6-localhost ip6-loopback
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
192.168.56.100 mycluster.icp

Output for lxc list
>vagrant@master:~$ lxc list
+---------+---------+-----------------------+------+------------+-----------+
|  NAME   |  STATE  |         IPV4          | IPV6 |    TYPE    | SNAPSHOTS |
+---------+---------+-----------------------+------+------------+-----------+
| worker1 | RUNNING | 192.168.56.101 (eth0) |      | PERSISTENT | 0         |
|               |                   | 172.17.0.1 (docker0)  |      |            |           |
|               |                   | 10.1.235.128 (tunl0)  |      |            |           |
+---------+---------+-----------------------+------+------------+-----------+
| worker2 | RUNNING | 192.168.56.102 (eth0) |      | PERSISTENT | 0         |
|                |                   | 172.17.0.1 (docker0)  |      |            |           |
|                |                   | 10.1.189.64 (tunl0)   |      |            |           |
+---------+---------+-----------------------+------+------------+-----------+

**Comment by mikespicer** [comment]
I have noticed that /etc/network/interfaces is referring to enp0s3 but ifconfig does not have an entry for enp0s3 (or enp0s8, but does have eth0 and eth1).  My suspicion is that the /etc/network/interfaces file is wrong but I am not sure what it should be.  Any ideas?

**Comment by tpouyer** [comment]
please verify that my latest changes have resolve this issue.  If so please close the issue. thx.

**Comment by DouglasGulledge** [comment]
I now have a working configuration, here are my observations, your mileage may vary.

- I didn't need to make ANY changes to a host file.  I did review a couple of Issues raised, but seemingly resolved, with pertinent changes to the Vagrantfile available in the latest code base.

- I leveraged the Vagrantfile "as is", with ONLY these three changes:
a) license = 'accept',  b) cpus = '6'  (was '4'),  c) memory = '12288' (was '10240'),
d) base_segment = '192.168.56' (was '192.168.27').

- (This seemed significant for me).  I rebooted the Mac after installing Virtualbox.  In debugging with a peer, this seemed necessary as for some reason the kernel changes weren't taking affect (until the reboot).  This was manifest in an inability for Virtualbox to update the MacOs network routing table.

- (MOST Important)  MacOS High Sierra now includes a Firewall.  System Pref > Security & Privacy.  I needed to make these two changes to the Firewall Policies:
a) Add Virtualbox so as to "Allow Incoming connections"
b) I disabled Stealth Mode to support test applications using ICMP, such as Ping.
